nsq/reader.py

```diff
@@ -677,7 +669,7 @@ def _redistribute_rdy_state(self):
             # We also don't attempt to avoid the connections who previously might have had RDY 1
             # because it would be overly complicated and not actually worth it (ie. given enough
             # redistribution rounds it doesn't matter).
-            possible_conns = list(self.conns.values())
+            possible_conns = [c for c in conns if not (c.in_flight or c.rdy)]
```

Comment:
The above last sentence should be: 

If we go from from `RDY 0` to `RDY 1` on a connection (other than the one just set to `RDY 0`) that has messages available for delivery, we guarantee a max-in-flight violation.